### PR TITLE
SDK-1833: third party identity

### PR DESCRIPTION
**Pull request by emmas-yoti**
### Added
 * ThirdPartyIdentityCheck as type to be returned as type for third party checks
 * ThirdPartyIdentityChecks() to GetSessionResult to retrieve third party check results
 * RequestedThirdPartyIdentityCheck type for adding a third party check to a sandbox session
 * ThirdPartyIdentityCheck type for configuring the expectation for a third party check in a sandbox session
 * WithThirdPartyIdentityCheck() to CheckReports for adding the expectation to a sandbox session

### CODE REVIEW

**Review by echarrod — CHANGES_REQUESTED** [review]
Please can we finish and merge https://github.com/getyoti/yoti-go-sdk/pull/208 first, so those changes don't appear in this PR too.

We also need to add this check to the example project. See https://github.com/getyoti/yoti-node-sdk/pull/259 for a reference


Also please add a description to each PR, even if it's just small. We might not need it now, but it might give some context if we need to look back at the PR. E.g.:
### Added
- Third Party Identity check as a possible type to be returned as check on the Doc Scan session
- Third Party Identity check to sandbox too

File: docscan/sandbox/request/task/document_text_data_extraction_task.go [hunk 1]
```diff
@@ -19,6 +19,7 @@ type DocumentTextDataExtractionTaskBuilder struct {
 	documentIDPhoto *documentIDPhoto
 	detectedCountry string
 	recommendation  *TextDataExtractionRecommendation
+	manualCheck     string
```
Comment (echarrod) [inline docscan/sandbox/request/task/document_text_data_extraction_task.go:22]: Do we need this line? 

Please can you make sure you install the pre-commit hook for every repo. 
Will be `pre-commit install` from the root, and `pre-commit run --all-files` to run. Running that from this branch gives me:
> docscan/sandbox/request/task/document_text_data_extraction_task.go:22:2: `manualCheck` is unused (structcheck)
        manualCheck     string

File: docscan/constants/constants.go [hunk 2]
```diff
@@ -7,9 +7,11 @@ const (
 	IDDocumentTextDataExtraction            string = "ID_DOCUMENT_TEXT_DATA_EXTRACTION"
 	IDDocumentFaceMatch                     string = "ID_DOCUMENT_FACE_MATCH"
 	SupplementaryDocumentTextDataCheck      string = "SUPPLEMENTARY_DOCUMENT_TEXT_DATA_CHECK"
+	ThirdPartyIdentityCheck                 string = "THIRD_PARTY_IDENTITY"
 	SupplementaryDocumentTextDataExtraction string = "SUPPLEMENTARY_DOCUMENT_TEXT_DATA_EXTRACTION"
 	Liveness                                string = "LIVENESS"
 	Zoom                                    string = "ZOOM"
+	SupplementaryDocumentType               string = "SUPPLEMENTARY_DOCUMENT"
```
Comment (echarrod) [inline docscan/constants/constants.go:14]: Doesn't look like this is used

File: docscan/sandbox/request/check/third_party_identity_check.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+type ThirdPartyIdentityCheck struct {
+	*check
+}
+
+// ThirdPartyIdentityCheckBuilder builds a ThirdPartyIdentityCheck
+type ThirdPartyIdentityCheckBuilder struct {
+	checkBuilder
+}
+
+func (c *ThirdPartyIdentityCheck) Report() checkReport {
```
Comment (echarrod) [inline docscan/sandbox/request/check/third_party_identity_check.go:15]: This is an exported method with an unexported type. I presume we want to export `checkReport` to solve this?

File: docscan/constants/constants.go (hunk 2 above)
Comment (emmas-yoti) [reply docscan/constants/constants.go:14]: It's used downstream in the sandbox repo.

**Comment by emmas-yoti** [comment]
> We also need to add this check to the example project. See [getyoti/yoti-node-sdk#259](https://github.com/getyoti/yoti-node-sdk/pull/259) for a reference

I've done the code for the third party check in the example project in a seperate PR to try to make it easier to review
https://github.com/getyoti/yoti-go-sdk/pull/210 specifically commit https://github.com/getyoti/yoti-go-sdk/pull/210/commits/d9a443a13f68b26200fd80feabf881704d6a6f85

**Review by echarrod — APPROVED** [review]

**Review by MrBurtyyy — APPROVED** [review]
Looks good to me :+1:

**Pull request merged**